week3/final/robot-text/robot.js

```diff
@@ -1,110 +1,123 @@
-        break;
-      case 'left':
-        x = x > 0 ? x - 1 : x;
-        break;
-      case 'right':
-        x = x < board[y].length - 1 ? x + 1 : x;
-        break;
-    }
+    'use strict';
+
+    const board = [
+        ['.', '.', '.', '.','.'],
+        ['.', '.', '.', '.','.'],
+        ['R', '.', '.', '.','F'],
+        ['.', 'A', 'A', 'A','.'],
+        ['.', '.', '.', '.','.']
+    ];
 
-    const cellContents = board[y][x];
+    const robot = {
+        x: 0,
+        y: 2,
+        dir: 'right'
+    };
 
-    if (cellContents === '.' || cellContents === 'F') {
-      board[robot.y][robot.x] = trailIndicators[robot.dir];
-      robot.x = x;
-      robot.y = y;
-      board[y][x] = 'R';
-      if (cellContents === 'F') {
-        flagReached = true;
-      }
+    let flagReached = false;
+    let appleeaten = false;
+    let appleseaten = 0;
+    let moves = 0;
+
+    board.reverse();
+
+    const trailIndicators = {
+        left: '←',
+        right: '→',
+        up: '↑',
+        down: '↓'
+    };
+
+    function render() {
+        console.log('\n ' + moves + ':');
+        for (let row = board.length - 1; row >= 0; row--) {
+            const cells = board[row];
+            let line = '';
+            for (let col = 0; col < cells.length; col++) {
+                line += ' ' + cells[col] + ' ';
+            }
+            console.log(line);
+        }
+        if (appleeaten) { console.log('\nYUM!');}
+        if (flagReached) {
+            console.log('\nHurray! Flag reached in ' + moves + ' steps!');
+            console.log('\n'+ appleseaten + 'apples are eaten!');
+        }
     }
 
-    moves += 1;
-    render();
-  }
+    function move() {
+        let x = robot.x;
+        let y = robot.y;
+        appleeaten = false;
 
-  function turn(turnDirection) {
-    if (turnDirection !== 'left' && turnDirection !== 'right') {
-      console.log('ignoring invalid turn', turnDirection);
+
+        switch (robot.dir) {
+            case 'up':
+                y = y < board.length - 1 ? y + 1 : y;
+                break;
+            case 'down':
+                y = y > 0 ? y - 1 : y;
+                break;
+            case 'left':
+                x = x > 0 ? x - 1 : x;
+                break;
+            case 'right':
+                x = x < board[y].length - 1 ? x + 1 : x;
+                break;
+        }
+
+        const cellContents = board[y][x];
+
+        if (cellContents === '.' || cellContents === 'F' || cellContents === 'A') {
+            board[robot.y][robot.x] = trailIndicators[robot.dir];
+            robot.x = x;
+            robot.y = y;
+            board[y][x] = 'R';
+            if (cellContents === 'F') {
+                flagReached = true;
+                
+            }
+            if (cellContents === 'A') {
+                appleseaten += 1;
+            }
+            if (appleseaten == 3) { appleeaten = true; }
+        }
+
+       moves += 1;
+        render();
```

Comment:
Your indentation is off here. You can press the three-fingered salute in VSCode that I mentioned [here](https://github.com/HackYourFuture/JavaScript/blob/master/fundamentals/code_formatting.md#use-vscode-for-well-formatted-javascript) to automatically correct indentation flaws.

Reply:
thank you for your comments, I tried to correct them.

Reply:
Hi Elmira, I still get only one Yum. I would expect one Yum for every apple you pass. It is also implied in the variable names I proposed:

```js
let appleEaten = false;
let applesEaten = 0;
```

Notice that the first variable is singular (apple) where as the second one is plural (apples). (Naming is extremely important in programming.) So the trick is to set the variable `appleEaten` to false at the start of the `move()` function and only set it to true if the cell contains an 'A' (irrespective of the number of apples already eaten). Then, in the `render()` function, say Yum if `appleEaten` is true.

Perhaps you didn't play the online game for this scenario? Because then you could have observed the same behaviour. 

The indentation flaws are corrected fine.